packages/ia-components/sandbox/theatres/with-youtube-spotify/utils/flatten-album-data.js

```diff
@@ -56,23 +58,31 @@ const flattenAlbumData = (metadata, playFullIAAudio) => {
     files: allFiles
   } = metadata;
   const { collection, identifier: itemIdentifier } = albumMetadata;
-  const fileDirectoryPrefix = `${server.includes('://') ? '' : 'https://'}${server}${directoryPath}/`
+  const fileDirectoryPrefix = `${server.includes('://') ? '' : 'https://'}${server}${directoryPath}/`;
 
   let countSampleMP3 = 0;
   let countFullMP3 = 0;
   let countOriginalAudioFiles = 0;
+  let countSegmentedTracks = 0;
   /**
    * Take original item's file list
    * & only return the files we are interested in
    */
   const slimFiles = allFiles.reduce((acc = [], file) => {
-    const { name: fileName, source = '' } = file;
+    const { name: fileName, source = '', original = '' } = file;
     const isValidAudio = isValidAudioFile(fileName);
     const isValidImage = isValidImageFile(fileName);
     const isNeededFile = isValidAudio || isValidImage;
     const fileIsDerived = source === 'derivative';
-    const isSampleMP3 = fileName.match('_sample.mp3');
-    const isMP3 = fileName.match('.mp3');
+    const isSampleMP3 = fileName.match(/_sample\.mp3$/);
+    const isMP3 = fileName.match(/\.mp3$/);
+    const isVBRMP3 = isOlderDerivedMP3(fileName);
+    const isFullMP3 = isMP3 && !isSampleMP3 && !isVBRMP3;
+    const isFullLengthTrack = isFullMP3 && fileIsDerived && !isVBRMP3;
```

Comment:
`isFullMP3` will be false if `!isVBRMP3` is false, so you can drop it from this short circuit.

Reply:
good eye!